pkgs/os-specific/bsd/freebsd/pkgs/include/package.nix

```diff
@@ -10,11 +11,15 @@ mkDerivation {
   noLibc = true;
   path = "include";
 
-  extraPaths = [
-    "contrib/libc-vis"
-    "etc/mtree/BSD.include.dist"
-    "sys"
-  ];
+  extraPaths =
+    [
+      "contrib/libc-vis"
+      "etc/mtree/BSD.include.dist"
+      "sys"
+    ]
+    ++ lib.optionals stdenv.isAarch64 [
+      "lib/msun/arm"
+    ];
```

Comment:
I wonder if this should be unconditional, for the same reason we make patches unconditional where possible? (So that contributors using a different architecture don't notice if e.g. something has moved around when updating.)

Reply:
We've been doing things conditionally to reduce rebuilds. It would also reduce the scope of the tree that would cause mass rebuilds if we make filtered source fixed-output (I had a prototype a while ago but it was more complexity than I wanted to merge int)

Reply:
Conditionals to reduce rebuilds are okay as long as they’re followed up by a `staging` PR that removes the condition.

Reply:
@emily oh this isn't about a temporary less rebuild, but a permanent one of some directories change but others don't.

Reply:
How likely is an update where only the ARM headers change but nothing else?

Reply:
Practically, not very high, Most of the changes I can find on releng branches (where security updates go) are for the kernel or tools. I'll remove the conditionals

Reply:
BTW, I don’t know if you’re at the point where you can make installer ISOs yet, and I’m not sure if you’re using filesets or similar here, but since it’s so rough to debug, fair warning that you might run into them being broken when you get to that point: https://github.com/NixOS/nixpkgs/pull/352491.

Reply:
We've been able to make VM images on x86_64-freebsd for quite a long time now. We haven't put any effort into ISOs though, when I installed on real hardware I wrote the VM image to a flash drive.
We don't use filesets, we just used rsync with paths from closureInfo, which I think the nixos upstream VM images use too, though I'm not sure.

Reply:
Oh, yeah, that bug, I've had issues with the installer on linux when referencing files relative to my modules. I don't recall having any more issues on freebsd than linux with that, though it was months ago.